core/development/Game.ts

```diff
@@ -104,12 +104,13 @@ function endMove(canRecover = true) {
 function checkForUpgradeablePawns() {
   for (let index = 0; index < pieces.length; index++) {
     const piece = pieces[index];
-    if (piece instanceof Pawn) {
+    if (piece instanceof Pawn && piece.position) {
       const whitePawnReachedEndOfBoard =
-        piece.player === whitePlayer && piece.position.coordinates[1] === 0;
+        piece.player === whitePlayer && piece.position?.coordinates[1] === 0;
```

Comment:
Do we need the ? after piece.position if the if already checks that piece.position is not undefined?

Reply:
We do not